lib/ClimaLandSimulations/src/utilities/pfts.jl

```diff
@@ -0,0 +1,614 @@
+        plant_S_s = 1e-2 * 0.0098, # Taken from site params
+        rooting_depth = 2.60,      # Bonan Table 2.3
+    )
+)
+
+
+C4G = Pft(
+    "C4 Grass",
+    (
+        # Radiative Transfer Model
+        Ω = 0.75,          # Chen et al. 2012 Table 1
+        α_PAR_leaf = 0.11, # CLM5.0 Table 2.3.1
+        α_NIR_leaf = 0.35, # CLM5.0 Table 2.3.1
+        τ_PAR_leaf = 0.05, # CLM5.0 Table 2.3.1
+        τ_NIR_leaf = 0.34, # CLM5.0 Table 2.3.1
+
+        # Canopy Energy Model
+        ac_canopy = 2.5e3, # Taken from site params
+
+        # Stomatal Conductance Model
+        g1 = 1.62, # CLM5.0 Table 2.9.1
+
+        # Photosynthesis Model,
+        Vcmax25 = 4.383, # Oliver et al. 2022 Table 2
+
+        # Hydraulics Model and Plant Parameters
+        SAI = 2.50,                # Guess
+        f_root_to_shoot = 3.50,    # Guess
+        K_sat_plant = 2.78e-6,     # Guess
+        ψ63 = -4.0 / 0.0098,       # Guess
+        Weibull_param = 4.00,      # Taken from site params
+        a = 0.05 * 0.0098,         # Taken from site params
+        capacity = 15.00,          # Guess
+        plant_S_s = 1e-2 * 0.0098, # Taken from site params
+        rooting_depth = 15.0,      # Bonan Table 2.3
+    )
+)
+
+# Lists of defaults to pass to parameter constructing function
+default_pfts = [
+    NET_Temp,
+    NET_Bor,
+    NDT_Bor,
+    BET_Trop,
+    BET_Temp,
+    BDT_Trop,
+    BDT_Temp,
+    BDT_Bor,
+    BES_Temp,
+    BDS_Temp,
+    BDT_Bor,
+    C3G_A,
+    C3G_NA,
+    C4G
+]
+
+
+function params_from_pfts(pft_cover::Vector{Float64}, 
+                            pfts::Vector{Pft}=default_pfts)
+    """
+    This function takes in a vector of PFT cover percentages and creates
+    variables for each parameter in the model that are the linear combination of
+    the PFT parameters weighted by the land cover percentages. For example,
+    after running this function the variable `Ω` will be the linear combination 
+    of the clumping indices of each PFT.
+
+    May optionally take in a vector of PFTs to use instead of the default PFTs.
+    In this case the pfts_cover vector must be the same length and in the
+    corresponding order to the PFTs in the pfts vector. This allows a user to 
+    define their own PFT scheme and plug it into the model.
+    """
+    # Construct the PFT DataFrame from the specified PFTs
+    pft_df = DataFrame()
+    for pft in pfts
+        # Push the internal NamedTuple of parameters to a row in the df. Since
+        # the PFT is stored as a NamedTuple conversion to a DF row is ensured 
+        # to work and ensure the ordering of the parameters.
+        push!(pft_df, pft.parameters)
+    end
+    
+    # Instantiate a variable with the correct name for the weighted average of 
+    # each parameter.
+    for param in param_list
+        # each param value is the dot product of the PFT cover percentages and
+        # the parameter values
+        param_value = pft_cover ⋅ pft_df[:, param]
```

Comment:
can change to just taking the params from the max(pft_cover) pft type as discussed today?